pyserini/trectools/_base.py

```diff
@@ -70,16 +86,26 @@ def get_docs_by_topic(self, topic: str, max_docs: int = None):
 
         return docs
 
-    def rescore(self, method: str, rrf_k: int):
-        if method == 'rrf':
+    def rescore(self, method: FusionMethod, rrf_k: int):
+        if method == FusionMethod.RRF:
             rows = []
 
             for topic, _, docid, rank, _, tag in self.run_data.to_numpy():
                 rows.append((topic, 'Q0', docid, rank, 1 / (rrf_k + rank), tag))
 
-            return TrecRun.get_trec_run_by_list(rows, self)
+            return TrecRun.get_trec_run_from_list(rows, self)
         else:
-            raise Exception(f'Unknown rescore method {method} detected')
+            raise NotImplementedError()
+
+    def map_trec_run_to_simple_search_result(self, docid_to_search_result: Dict[str, JSimpleSearcherResult]) -> List[JSimpleSearcherResult]:
```

Comment:
map -> convert: save map for the technical functional programming sense.

just `convert_to_search_results`? `trec_run` is implict give it's a method?

Reply:
On second thought, this method should be directly inside the fusion searcher... because we're passing in `Dict[str, JSimpleSearcherResult]`, breaks separation of concerns since `TrecRun` doesn't deal with a result object anywhere else?